### DISCUSSION TITLE
Deploy using symbolic links instead of copying files

### DISCUSSION THREAD

**Discussion by mkimberling** [pyinstaller/pyinstaller#7202] [Features & Development]
A suggestion: A new feature such that in folder mode, files are deployed to the dist folder with a symbolic link instead of a copy of the files.

Context: the footprint of my PyInstaller application is currently 1gb and it takes 3+ minutes to build it.  I suspect iterating on my app would be faster if it wasn't writing these huge files over and over.  It wouldn't hurt to reduce the aging of my ssd that happens from copying these files every time I build it.

**Comment by bwoodsend** [comment]
But then it wouldn't be distributable. As soon as you copy the `dist` folder to another computer, all those symlinks would be broken. Likewise if you move the `dist` folder or modify/delete the environment you used to build it.

**Reply by mkimberling** [reply]
The primary use of an option that would use symlinks is just faster development iteration.  The build of my app I distribute to users is created by CI servers.  They would never use the option.

**Reply by bwoodsend** [reply]
Copying files isn't the bottleneck in PyInstaller. SSDs can write >100MB/s these days so copying 1GB of files is only going to be responsible for a few seconds of overhead. It's more the dependency scanning and in onefile mode, zip compressing everything. Doing brute force collection (i.e. using `--collect-all`) will make that much worse. If you're really concerned about the life of your SSD then you could build your application in `/tmp` which is mounted in RAM.
